source/common/config/subscription_factory.h

```diff
@@ -54,6 +60,11 @@ class SubscriptionFactory {
             "envoy::api::v2::ConfigSource must have a singleton cluster name specified");
       }
       const auto& cluster_name = api_config_source.cluster_name()[0];
+      if (!cm.get(cluster_name)) {
```

Comment:
I think you need to additionally check that this cluster was not "added via API" (https://github.com/envoyproxy/envoy/blob/master/include/envoy/upstream/upstream.h#L333). I think by definition the cluster below can't be added by API but you might assert that.

Reply:
I was thinking about this on my bike ride home yesterday! We definitely need to add that additional check in order to accomplish what I set out to enforce with this checks